apps/prairielearn/src/lib/db-types.ts

```diff
@@ -480,3 +480,88 @@ export const InstanceQuestionSchema = z.object({
   variants_points_list: z.array(z.number()),
 });
 export type InstanceQuestion = z.infer<typeof InstanceQuestionSchema>;
+
+export const SubmissionSchema = z.object({
+  auth_user_id: IdSchema.nullable(),
+  broken: z.boolean().nullable(),
+  correct: z.boolean().nullable(),
+  credit: z.number().nullable(),
+  date: DateFromISOString.nullable(),
+  duration: IntervalSchema.nullable(),
+  feedback: z.record(z.string(), z.any()).nullable(),
+  format_errors: z.record(z.string(), z.any()).nullable(),
+  gradable: z.boolean().nullable(),
+  graded_at: DateFromISOString.nullable(),
+  grading_method: z.enum(['Internal', 'External', 'Manual']).nullable(),
+  grading_requested_at: DateFromISOString.nullable(),
+  id: IdSchema,
+  manual_rubric_grading_id: IdSchema.nullable(),
+  mode: z.enum(['Public', 'Exam', 'SEB']).nullable(),
+  override_score: z.number().nullable(),
+  params: z.record(z.string(), z.any()).nullable(),
+  partial_scores: z.record(z.string(), z.number()).nullable(),
+  raw_submitted_answer: z.record(z.string(), z.any()).nullable(),
+  regradable: z.boolean().nullable(),
+  score: z.number().nullable(),
+  sid: z.string().nullable(),
```

Comment:
@nwalters512 should I remove this given #8827?

Reply:
@jonatanschroeder do it once that lands, otherwise I think you may start seeing test failures here.

Reply:
@nwalters512 there won't be errors there because (1) this schema is not used by that query, and (2) that query adds a column, but the column's value is not used. Zod strips columns that exist in the query result but are not in the schema, since we're not [using strict mode](https://github.com/colinhacks/zod#strict).

Reply:
Indeed! I had things backwards.

Reply:
Done in 66e13155a5ebfe041e915c02d57501488c1ea46c. Also removed qiid from variant. The other columns listed in #8828 are from assessment_instances which doesn't have a schema yet.